eva_vcf_merge/merge.py

```diff
@@ -26,31 +28,54 @@ def __init__(self, bgzip_binary, bcftools_binary, nextflow_binary, nextflow_conf
         self.nextflow_binary = nextflow_binary
         self.nextflow_config = nextflow_config
         self.output_dir = output_dir
+        self.working_dir = os.path.join(output_dir, 'nextflow')
 
-    def horizontal_merge(self, vcf_groups):
+    def horizontal_merge(self, vcf_groups, resume=True):
         """
         Merge groups of vcfs horizontally, i.e. by sample, using bcftools.
 
         :param vcf_groups: dict mapping a string (e.g. an analysis alias) to a group of vcf files to be merged
+        :param resume: whether to resume pipeline (default true)
         :returns: dict of merged filenames
         """
         pipeline, merged_filenames = self.generate_horizontal_merge_pipeline(vcf_groups)
-        workflow_file = os.path.join(self.output_dir, 'merge_workflow.nf')
+        workflow_file = os.path.join(self.working_dir, 'horizontal_merge.nf')
+        os.makedirs(self.working_dir, exist_ok=True)
         pipeline.run_pipeline(
             workflow_file_path=workflow_file,
-            working_dir=self.output_dir,
+            working_dir=self.working_dir,
             nextflow_binary_path=self.nextflow_binary,
-            nextflow_config_path=self.nextflow_config
+            nextflow_config_path=self.nextflow_config,
+            resume=resume
         )
         return merged_filenames
 
-    def vertical_concat(self, vcf_groups):
+    def vertical_merge(self, vcf_groups, chunk_size=500, resume=True):
         """
         Merge groups of vcfs vertically, i.e. concatenation.
 
         :param vcf_groups: dict mapping a string (e.g. an analysis alias) to a group of vcf files to be merged
+        :param chunk_size: number of vcfs to merge at once (default 500)
+        :param resume: whether to resume pipeline (default true)
+        :returns: dict of merged filenames
         """
-        raise NotImplementedError('Vertical concatenation not yet implemented.')
+        pipeline, merged_filenames = self.generate_vertical_merge_pipeline(vcf_groups, chunk_size)
+        workflow_file = os.path.join(self.working_dir, "vertical_concat.nf")
+        os.makedirs(self.working_dir, exist_ok=True)
+        pipeline.run_pipeline(
+            workflow_file_path=workflow_file,
+            working_dir=self.working_dir,
+            nextflow_binary_path=self.nextflow_binary,
+            nextflow_config_path=self.nextflow_config,
+            resume=resume
+        )
+        # move merged files to output directory and rename with alias
+        for alias in merged_filenames:
+            safe_alias = get_safe_str(alias)
```

Comment:
being a bit paranoid here: there is a risk of name collision between multiple aliases that might only differ by some special character. I have seen things being name like `samples(+)` and `samples(-)` which would both translate to `samples___`

Reply:
That's true, I think the entire method should probably throw an error if two aliases would map to the same filename (and the docstring make clear that that's one thing the alias is being used for).

For actually dealing with these collisions I'm not sure the best thing to do.  From the perspective of the submission automation, it's kind of overloading the purpose of the analysis alias to use it as a filename, but there's not much else to use since the files are pre-brokering and will be archived with whatever name we give them...

Reply:
Created [EVA-2560](https://www.ebi.ac.uk/panda/jira/browse/EVA-2560) to check this in validation in submission automation.